persistence/apikeys.go

```diff
@@ -0,0 +1,34 @@
+package persistence
+
+import "sync"
+
+// Data structure to store API keys
+var apiKeys struct {
+	sync.RWMutex
+	keys map[string]bool
+}
+
+func init() {
+	apiKeys.keys = make(map[string]bool)
+}
+
+// AddAPIKey adds a new API key to the store
+func AddAPIKey(key string) {
+	apiKeys.Lock()
+	defer apiKeys.Unlock()
+	apiKeys.keys[key] = true
+}
+
+// DeleteAPIKey removes an API key from the store
+func DeleteAPIKey(key string) {
+	apiKeys.Lock()
+	defer apiKeys.Unlock()
+	delete(apiKeys.keys, key)
```

Comment:
What if the key does not exists? What happens? We need to implement a check that the key exists. If not return a error

Reply:
When we move the datastructure to a slice, use something like the following:
https://stackoverflow.com/questions/37334119/how-to-delete-an-element-from-a-slice-in-golang